lib/index.d.ts

```diff
@@ -0,0 +1,124 @@
+import type { Boom } from "@hapi/boom";
+import type { AssertionError } from "assert";
+
+export type BounceErrorType = Error | "system" | "boom" | Record<any, any>;
```

Comment:
`Record<any, any>` does not seem appropriate, as it is intended to [construct an object type using keys from another object](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type). I would do a simple: `{ [key: PropertyKey]: any }`.